### PR TITLE
Don't swallow error

### PR DESCRIPTION
**Pull request by benrudolph**
@czue 
elf: @esoergel 

This will hopefully help with errors bugs like this one: http://manage.dimagi.com/default.asp?167785. I found out that `cases.js` was throwing the same error no matter what the actual error was on touchforms

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/backbone/cases.js [hunk 1]
```diff
@@ -4,9 +4,10 @@ if (typeof cloudCare === 'undefined') {
 
 cloudCare.EMPTY = '---';
 
-var _caseListLoadError = function (response) {
+var _caseListLoadError = function (model, response) {
     hideLoadingCallback();
-    showError(translatedStrings.caseListError, $("#cloudcare-notifications"));
+    var message = response.responseText || translatedStrings.caseListError
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/cases.js]: Missing semicolon.

**Comment by benrudolph** [comment]
now thinking of it, you may have done this on purpose. the error message probably won't me as nice as the response from touchforms

**Comment by czue** [comment]
@benrudolph I don't remember exactly which error this is, but I think I did this to make it less confusing for users.

**Comment by benrudolph** [comment]
Sounds good. It's the one that gets triggered when an error happens when filtering the case list. What do you think about console.error for the actual error. It's pretty useful for debugging and I'm going to be working on fixing error messages in touch forms. Currently the only way to see the error that I know of is to actually log onto the prod server

**Comment by czue** [comment]
yeah console.error seems super userful

**Comment by benrudolph** [comment]
:heavy_check_mark:

File: corehq/apps/cloudcare/static/cloudcare/js/backbone/cases.js [hunk 2]
```diff
@@ -4,8 +4,9 @@ if (typeof cloudCare === 'undefined') {
 
 cloudCare.EMPTY = '---';
 
-var _caseListLoadError = function (response) {
+var _caseListLoadError = function (model, response) {
```
Comment (czue) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/cases.js]: sorry if this should be obvious, but why is the signature changing? was it just wrong?

Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/backbone/cases.js]: yea, backbone usually passes model as first arg: http://backbonejs.org/#Model-fetch. 
https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/static/hqwebapp/js/lib/backbone.js#L1226. also tested this via a breakpoint

**Pull request merged**